examples/wifi/expanded/requests_wifi_api_openskynetwork_private.py

```diff
@@ -19,135 +18,177 @@
 # All active flights JSON: https://opensky-network.org/api/states/all  # PICK ONE! :)
 # JSON order: transponder, callsign, country
 # ACTIVE transpondes only, for multiple "c822af&icao24=cb3993&icao24=c63923"
-transponder = "7c6b2d"
+TRANSPONDER = "471efd"
 
-# Initialize WiFi Pool (There can be only 1 pool & top of script)
-pool = socketpool.SocketPool(wifi.radio)
+# Get WiFi details, ensure these are setup in settings.toml
+ssid = os.getenv("CIRCUITPY_WIFI_SSID")
+password = os.getenv("CIRCUITPY_WIFI_PASSWORD")
+osnusername = os.getenv("OSN_USERNAME")  # Website Credentials
+osnpassword = os.getenv("OSN_PASSWORD")  # Website Credentials
 
-# Time between API refreshes
+# API Polling Rate
 # 900 = 15 mins, 1800 = 30 mins, 3600 = 1 hour
 # OpenSky-Networks IP bans for too many requests, check rate limit.
 # https://openskynetwork.github.io/opensky-api/rest.html#limitations
-sleep_time = 1800
+SLEEP_TIME = 1800
 
-# Get WiFi details, ensure these are setup in settings.toml
-ssid = os.getenv("CIRCUITPY_WIFI_SSID")
-password = os.getenv("CIRCUITPY_WIFI_PASSWORD")
-osnu = os.getenv("OSN_Username")
-osnp = os.getenv("OSN_Password")
-
-osn_cred = str(osnu) + ":" + str(osnp)
-bytes_to_encode = b" " + str(osn_cred) + " "
-base64_string = base64.encodebytes(bytes_to_encode)
-base64cred = repr(base64_string)[2:-1]
-
-Debug_Auth = False  # STREAMER WARNING this will show your credentials!
-if Debug_Auth:
-    osn_cred = str(osnu) + ":" + str(osnp)
-    bytes_to_encode = b" " + str(osn_cred) + " "
-    print(repr(bytes_to_encode))
-    base64_string = base64.encodebytes(bytes_to_encode)
-    print(repr(base64_string)[2:-1])
-    base64cred = repr(base64_string)[2:-1]
-    print("Decoded Bytes:", str(base64cred))
+# Set debug to True for full JSON response.
+# WARNING: makes credentials visible
+DEBUG = False
+
+# Initalize Wifi, Socket Pool, Request Session
+pool = adafruit_connection_manager.get_radio_socketpool(wifi.radio)
+ssl_context = adafruit_connection_manager.get_radio_ssl_context(wifi.radio)
+requests = adafruit_requests.Session(pool, ssl_context)
+
+# -- Base64 Conversion --
+OSN_CREDENTIALS = str(osnusername) + ":" + str(osnpassword)
+OSN_CREDENTIALS_B = b"" + str(OSN_CREDENTIALS) + ""
+BASE64_ASCII = b2a_base64(OSN_CREDENTIALS_B)
+BASE64_STRING = str(BASE64_ASCII)  # bytearray
+TRUNCATED_BASE64_STRING = BASE64_STRING[2:-1]  # truncate bytearray head/tail
```

Comment:
Is this line's intention to cut out the `b'` at the beginning and `'` at the end of a bytestring representation to basically "convert" it to a regular string?

It seems like that is what it's doing to me, if so I think there are better ways to convert between string and bytestring without having to replace those "manually" like this. If my understanding of the intention is correct I'll figure out the specific syntax for the other ways to make that conversion.

Reply:
Yes that was the intention for the base64 string to just be a string because it needs to be inserted into the POST as a concatenated url. I think there's a way to use .decode possibly but manually slicing works. I tested each one and each iteration prior to submitting.  I'll have to take a look into why it's not slicing properly.

The private api's require website login and provide a higher daily call limit. The public API's do not require a login or token.  I might not have named the files the best names for this.  I just made them up as I went.  If you think there's a better naming convention for these API examples please feel free to rename them. 

Yes I intended to keep private and public as separate branches and accidentally committed 2 PRs from separate branches to the same branch. :(  It's definitely a problem that needs to be resolved. I can probably do it by deleting the branches and remaking them unless you have a better way.  

I do not have a code backup so please don't delete them.  I've had 3 hours of sleep and running on empty.  I can attempt to resolve tomorrow.